pkg/compact/compact.go

```diff
@@ -1160,6 +1160,11 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 					return errors.Wrapf(err,
 						"block id %s, try running with --debug.accept-malformed-index", meta.ULID)
 				}
+
+				if err := stats.DuplicatedSampleErr(); err != nil {
+					return errors.Wrapf(err, "block with duplicated samples found %s; Compaction level %v; Labels: %v", bdir, meta.Compaction.Level, meta.Thanos.Labels)
```

Comment:
You need to use `fmt.Sprintf()` inside it.
And I feel it is unnecessary to include compaction level and labels information in th error message itself.